solr/core/src/test/org/apache/solr/update/processor/UpsertConditionTest.java

```diff
@@ -178,6 +180,87 @@ public void givenNumericField_whenMatching() {
     assertTrue(condition.matches(oldDoc, newDoc));
   }
 
+  @Test
+  public void givenAtomicUpdateSet_whenMatching() {
+    NamedList<String> args = new NamedList<>(ImmutableMap.of(
+        "must", "OLD.field:value",
+        "action", "skip"
+    ));
+
+    UpsertCondition condition = UpsertCondition.parse("skip-it", args);
+
+    assertThat(condition.getName(), is("skip-it"));
+
+    SolrInputDocument oldDoc = new SolrInputDocument();
+    SolrInputDocument newDoc = new SolrInputDocument();
+
+    oldDoc.setField("field", Collections.singletonMap("set", "other1"));
```

Comment:
Kind of strange, because old docs (i.e. ones in the index) will never be atomic updates